src/app/common/context/MustGatherContext.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import { useMustGathersQuery } from '@app/queries';
+import { UseQueryResult } from 'react-query';
+import { IMustGatherResponse, mustGatherStatus } from '@app/client/types';
+import { MustGatherWatcher } from '@app/common/components/MustGatherWatcher';
+import { AppContext } from '@app/common/context/AppContext';
+
+export type mustGatherObjType = {
+  customName: string;
+  type: 'plan' | 'vm';
+  status: mustGatherStatus;
+};
+
+export type mgWatchListType = {
+  name: string;
+  isGathering: boolean;
+}[];
+
+export type mustGatherListType = mustGatherObjType[];
+interface IMustGatherContext {
+  mustGatherModalOpen: boolean;
+  setMustGatherModalOpen: (isOpen: boolean) => void;
+  mustGatherList: mustGatherListType;
+  setActiveMustGather: (mustGatherObj: mustGatherObjType) => void;
+  activeMustGather: mustGatherObjType | undefined;
+  mustGatherWatchList: mgWatchListType;
+  setMustGatherWatchList: (list: mgWatchListType) => void;
+  mustGathers: UseQueryResult<IMustGatherResponse[], Response>;
+}
+
+const mustGatherContextDefaultValue = {} as IMustGatherContext;
+export const MustGatherContext = React.createContext<IMustGatherContext>(
+  mustGatherContextDefaultValue
+);
+
+interface IMustGatherContextProvider {
+  children: React.ReactNode;
+}
+
+export const MustGatherContextProvider: React.FunctionComponent<IMustGatherContextProvider> = ({
+  children,
+}: IMustGatherContextProvider) => {
+  const appContext = React.useContext(AppContext);
+  const [mustGatherModalOpen, setMustGatherModalOpen] = React.useState(false);
+  const [mustGatherList, setMustGatherList] = React.useState<mustGatherListType>([]);
+  const [activeMustGather, setActiveMustGather] = React.useState<mustGatherObjType>();
+  const [mustGatherWatchList, setMustGatherWatchList] = React.useState<mgWatchListType>([]);
+  const [errorCount, setErrorCount] = React.useState(0);
```

Comment:
Can we make this a boolean like `errorNotified` or something? It's fine if you want to leave it as a number, it just makes me want to scan the code to see what happens when errors add up.